src/corporacreator/preprocessors/cy.py

```diff
@@ -1,12 +1,38 @@
-def cy(client_id, sentence):
-    """Cleans up the passed sentence, removing or reformatting invalid data.
+# compile the set of the alphabet just once
+alphabet = set([' ', 'a','á', 'à', 'â', 'ä', 'b', 'c',
+                'd', 'e', 'é', 'è', 'ê', 'ë', 'f', 'g',
+                'h', 'i', 'í', 'ì', 'î', 'ï', 'j', 'l',
+                'm', 'n', 'o', 'ó', 'ò', 'ô', 'ö', 'p',
+                'r', 's', 't', 'u', 'ú', 'ù', 'û', 'ü',
+                'w', 'ẃ', 'ẁ', 'ŵ', 'ẅ', 'y', 'ý', 'ỳ',
+                'ŷ', 'ÿ'])
 
+def cy(client_id, sentence):
+    """
+    Cleans up the passed sentence, removing or reformatting invalid data.
     Args:
       client_id (str): Client ID of sentence's speaker
       sentence (str): Sentence to be cleaned up.
+      alphabet (set): set of unique, lowercased characters in language.
 
     Returns:
       (str): Cleaned up sentence. Returning None or a `str` of whitespace flags the sentence as invalid.
     """
-    # TODO: Clean up cy data
+    # TODO: geiriau Saesneg / English inspired/pronunced words:
+    # wallace, celsius, ddiesel, wicipedia, william, chiswell, f., h.
+
+    # this if loop will skip the following else loop in the case that
+    # the sentence is purely Welsh characters as per https://github.com/
+    # unicode-cldr/cldr-misc-full/blob/master/main/cy/characters.json
+    if set(sentence.lower()).issubset(alphabet):
+        pass
+    else:
+        # Not all chars are in the Welsh alphabet!
+        # more processing needed here!
+        sentence = sentence.replace(",", "") # remove commas
+        sentence = sentence.replace("’", "'") # fix apostrophes
+        sentence = sentence.replace("wwna", "wna")
```

Comment:
All characters of "wwna" are in the alphabet, but this is only run on sentence if not all characters in the sentence are in alphabet.

So this is only run on sentences that have character not in alphabet **_and_** also contain the string "wwna".  So in particular, it does not have same effect as Dewi's code.

Reply:
@kdavis-mozilla this is an issue, indeed... suggestions?

Reply:
Move this to the other branch of the else and remove the pass.

Reply:
resolved in latest commit, no alphabet checking loop